### PR TITLE
UI rearrangement

### PR DESCRIPTION
**Pull request by TimMeiwald**

### CODE REVIEW

**Comment by dgtlrst** [comment]
@TimMeiwald here is my review:
1. Some file names are duplicating and it makes it a bit confusing which is what. For example, you have 3 file names, which contain the word `terminal`:
1. terminal_IO_tab.dart
2. terminal_settings_tab.dart
3. terminal.dart
It's not very clear what `terminal.dart` is responsible for: either put a comment inside the file or rename the file in the style of the rest two.

2. There is a general lack of comments in the code, it would be easier for other people to orient themselves in the code, if there were some comments here and there either around hard busingess logic or at creation/initialization of key elements.

3. Also - and if that's not too much of a bother - please mark clearly (can do with a comment again), the interface to the core rust library which the backend will be providing (e.g this currently happens to be `main()` routine of `TerminalRunner` class.

**Comment by dgtlrst** [comment]
Additionally, @TimMeiwald, when i built the project with flutter, it notified me that there exist issues that can be fixed with the command `flutter fix`.

**Comment by dgtlrst** [comment]
Also, there are dependencies with old versions and it complains about it. Check if some of them can be updated, including `flutter_to_rust_bridge`

File: lib/home.dart [hunk 1]
```diff
@@ -1,60 +1,110 @@
...
+  // var grid = GridView.count(
+  //   shrinkWrap: true,
+  //   padding: const EdgeInsets.all(10),
+  //   childAspectRatio: 16 / 9,
+  //   mainAxisSpacing: 10, // Adjust vertical spacing
+  //   crossAxisSpacing: 10, // Adjust horizontal spacing
+  //   crossAxisCount: 2,
+  //   children: terminals,
+  // );
+  // return Expanded(child: grid);
```
Comment (dgtlrst) [inline lib/home.dart:107]: remove unused commented code if it is not needed

File: lib/settings.dart [hunk 2]
```diff
@@ -1,225 +1,96 @@
 import 'dart:ffi';
 
+import 'package:bit/State_globals.dart';
 import 'package:bit/src/rust/api/serial.dart';
+import 'package:bit/State_app.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter/widgets.dart';
 
 import 'sidepanel.dart'; // side panel
```
Comment (dgtlrst) [inline lib/settings.dart:9]: this comment is kinda unnecessary

File: rust/Cargo.toml [hunk 3]
```diff
@@ -9,4 +9,7 @@ crate-type = ["cdylib", "staticlib"]
 [dependencies]
 anyhow = "1.0.86"
 flutter_rust_bridge = "=2.1.0"
```
Comment (dgtlrst) [inline rust/Cargo.toml:11]: i am getting an error from rust_analyzer here, saying we are 2 minor versions behind latest (2.3.0), would it be breaking changes if we updated this?

File: rust/src/api/serial.rs [hunk 4]
```diff
@@ -98,6 +105,33 @@ impl SerialPortInfo {
...
+        return match serial_port_info_as_str {
+            Ok(valid) => {
+                return Some(valid);
+            }
+            Err(e) => {
+                println!("{:?}", e);
+                None
+            }
+        };
+    }
```
Comment (dgtlrst) [inline rust/src/api/serial.rs:134]: where are these methods used and what are they needed for?

**Review by dgtlrst — CHANGES_REQUESTED** [review]
here are some comments

File: rust/src/api/serial.rs (hunk 4 above)
Comment (TimMeiwald) [reply rust/src/api/serial.rs:134]: They're used to serialize the port settings so I can save them using the shared_preferences lib that uses the platform specific app directory to save data. It's used in state_persistent_datastore.

File: rust/Cargo.toml (hunk 3 above)
Comment (TimMeiwald) [reply rust/Cargo.toml:11]: Apparently not, I'll make a new commit

**Pull request closed without merging**